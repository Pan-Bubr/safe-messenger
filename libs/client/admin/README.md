# client-admin

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test client-admin` to execute the unit tests.
