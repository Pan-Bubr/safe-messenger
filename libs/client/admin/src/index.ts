export * from './lib/client-admin.module';
