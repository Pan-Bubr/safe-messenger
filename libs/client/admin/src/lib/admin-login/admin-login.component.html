<div class="pt-5 pb-5">
  <form [formGroup]="loginForm" (ngSubmit)="login(loginForm.value)">
    <div *ngIf="unauthorizedError" class="alert alert-danger" role="alert">
      <h4 class="alert-heading">Unauthorized Error!</h4>
      <p class="mb-0">Your credentials are invalid</p>
    </div>

    <div class="input-group mb-3">
      <div class="input-group-prepend">
        <span class="input-group-text">Login: </span>
      </div>

      <input
        name="login"
        formControlName="login"
        type="text"
        class="form-control"
        placeholder="Enter login..."
      />
    </div>

    <div class="input-group mb-3">
      <div class="input-group-prepend">
        <span class="input-group-text">Password: </span>
      </div>

      <input
        name="password"
        formControlName="password"
        type="password"
        class="form-control"
        placeholder="Enter password..."
      />
    </div>

    <div class="text-center">
      <button
        class="btn btn-lg btn-secondary"
        type="submit"
        [disabled]="!loginForm.valid"
      >
        Login
      </button>
    </div>
  </form>
</div>
