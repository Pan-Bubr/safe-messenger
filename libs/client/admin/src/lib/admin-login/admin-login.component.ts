import { HttpClient, HttpErrorResponse } from '@angular/common/http';
import { Component } from '@angular/core';
import { FormControl, FormGroup, Validators } from '@angular/forms';
import { Router } from '@angular/router';

import { AdminAuthService } from '../services/admin-auth.service';

@Component({
  selector: 'safe-messenger-admin-login',
  templateUrl: './admin-login.component.html',
  styleUrls: ['./admin-login.component.scss']
})
export class AdminLoginComponent {
  constructor(
    private readonly adminAuthService: AdminAuthService,
    private readonly router: Router
  ) {}

  public loginForm: FormGroup = new FormGroup({
    login: new FormControl('', Validators.required),
    password: new FormControl('', Validators.required)
  });
  public unauthorizedError: boolean;

  public login({ login, password }: { login: string; password: string }): void {
    this.adminAuthService.login(login, password).subscribe(
      () => {
        return this.router.navigate(['admin', 'panel']);
      },
      (err: HttpErrorResponse) => {
        this.unauthorizedError = err.status === 401;
        this.loginForm = new FormGroup({
          login: new FormControl(),
          password: new FormControl()
        });
      }
    );
  }
}
