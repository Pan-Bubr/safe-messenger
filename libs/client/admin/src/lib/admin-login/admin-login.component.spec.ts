import { RouterTestingModule } from '@angular/router/testing';
import { ClientAdminModule } from '@safe-messenger/client/admin';

import { of, throwError } from 'rxjs';
import { Shallow } from 'shallow-render';

import { AdminAuthService } from '../services/admin-auth.service';
import { AdminLoginComponent } from './admin-login.component';

describe('LoginContainerComponent', () => {
  let shallow: Shallow<AdminLoginComponent>;

  beforeEach(() => {
    shallow = new Shallow(AdminLoginComponent, ClientAdminModule).import(
      RouterTestingModule
    ).provideMock({
      provide: AdminAuthService,
      useValue: {
        login: jest.fn(() => of(''))
      }
    });
  });

  it('should match snapshot', async () => {
    const { fixture } = await shallow.render();

    expect(fixture).toMatchSnapshot();
  });

  it('login() should send login request', async () => {
    const { instance, get } = await shallow.render();

    instance.login({
      login: 'test',
      password: 'testPassword'
    });

    expect(get(AdminAuthService).login).toBeCalled();
  });

  it('login() should send login request', async () => {
    const { instance, get } = await shallow.render();

    (get(AdminAuthService).login as jest.Mock).mockImplementationOnce(throwError);

    instance.login({
      login: 'test',
      password: 'testPassword'
    });

    expect(get(AdminAuthService).login).toBeCalled();
  });
});
