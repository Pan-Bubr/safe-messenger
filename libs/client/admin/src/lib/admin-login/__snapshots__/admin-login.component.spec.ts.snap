// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainerComponent should match snapshot 1`] = `
<ng-component>
  <safe-messenger-admin-login>
    <div
      class="pt-5 pb-5"
    >
      <form
        class="ng-untouched ng-pristine ng-invalid"
        novalidate=""
      >
        
        <div
          class="input-group mb-3"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              Login: 
            </span>
          </div>
          <input
            class="form-control ng-untouched ng-pristine ng-invalid"
            formcontrolname="login"
            name="login"
            placeholder="Enter login..."
            type="text"
          />
        </div>
        <div
          class="input-group mb-3"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              Password: 
            </span>
          </div>
          <input
            class="form-control ng-untouched ng-pristine ng-invalid"
            formcontrolname="password"
            name="password"
            placeholder="Enter password..."
            type="password"
          />
        </div>
        <div
          class="text-center"
        >
          <button
            class="btn btn-lg btn-secondary"
            disabled=""
            type="submit"
          >
             Login 
          </button>
        </div>
      </form>
    </div>
  </safe-messenger-admin-login>
</ng-component>
`;
